### ISSUE TITLE
Error: null

### ISSUE THREAD

**Issue by harshal2030** [souvik-ghosh/react-native-create-thumbnail#16]
Hi friend. I am using react-native-image-picker to pick videos.
Here is the code
```javascript
createThumbnail({
                    url: response.uri,
                    timeStamp: 1000,
                    type: 'local',
                    format: 'png',
                })
                    .then(resp => console.log(resp))
                    .catch(e => console.log(e));
```
And i keep getting this error
![error](https://user-images.githubusercontent.com/43959510/80926128-d8402900-8db2-11ea-8968-88bf919ed87c.png)

**Comment by souvik-ghosh** [comment]
@harshal2030 Are you getting this in both android and ios?

**Comment by harshal2030** [comment]
Well I don't have ios, I am getting this on Android

**Comment by zacwasielewski** [comment]
I'm also getting this on Android. Works fine on iOS.

**Comment by souvik-ghosh** [comment]
@harshal2030 @zacwasielewski  Thanks for the inputs, I was able to reproduce this error. This is occurring in case the `url` provided is invalid. Are you able to play the video using the same path?

In the next version, a more detailed error message will be provided instead of 'null'.

**Comment by harshal2030** [comment]
Yes video is playing with same url

**Comment by souvik-ghosh** [comment]
@harshal2030 Can you please test this with the latest version?

**Comment by harshal2030** [comment]
Okk, I'll back to you within tommorow

**Comment by ar0692** [comment]
I am having issue in android, ios its working perfectly 


ExceptionsManager.js:173 Unhandled promise rejection Error: setDataSource failed: status = 0xFFFFFFEA
    at promiseMethodWrapper (NativeModules.js:103)
    at createThumb$ (AddPostScreen.js:41)
    at tryCatch (runtime.js:45)
    at Generator.invoke [as _invoke] (runtime.js:274)
    at Generator.prototype.<computed> [as next] (runtime.js:97)
    at tryCatch (runtime.js:45)
    at invoke (runtime.js:135)
    at runtime.js:170
    at new Promise (es.promise.js:233)
    at callInvokeWithMethodAndArg (runtime.js:169)

**Comment by ar0692** [comment]
I want to show duration time of video and video symbol on thumbnail 
can we do that? or any work around it

**Comment by toggm** [comment]
I see the same issue in a project of mine. File exists, permissions are correct. Error doesn't say that much.

**Comment by IcedElect** [comment]
+1

**Comment by Janak-Nirmal** [comment]
Error is at this function

```
  private static Bitmap getBitmapAtTime(String filePath, int time, Map headers) {
        MediaMetadataRetriever retriever = new MediaMetadataRetriever();
        if (URLUtil.isFileUrl(filePath)) {
            retriever.setDataSource(Uri.decode(filePath).replace("file://", ""));
        } else {
            if (VERSION.SDK_INT < 14) {
                throw new IllegalStateException("Remote videos aren't supported on sdk_version < 14");
            }
            retriever.setDataSource(filePath, headers);
        }
  
        Bitmap image = retriever.getFrameAtTime(time * 1000, MediaMetadataRetriever.OPTION_CLOSEST_SYNC);
        retriever.release();
        if (image == null) {
            throw new IllegalStateException("File doesn't exist or not supported");
        }
        return image;
    }
```

If you are on latest react-native-image-picker version response is coming like below

{"fileName": "VID-20210112-WA0015.mp4", "fileSize": 2251360, "uri": "content://media/external/video/media/66056"}

and it goes at,

```
retriever.setDataSource(filePath, headers); 
```
and it gives error.

@souvik-ghosh If you can  assist what to change in this I believe this issue will be resolved.

**Comment by Janak-Nirmal** [comment]
Below is modified version of the getBitMapAtTime function which works as of now. 

```
    private Bitmap getBitmapAtTime(String filePath, int time, Map headers) {
        MediaMetadataRetriever retriever = new MediaMetadataRetriever();
        retriever.setDataSource(this.reactContext,Uri.parse(filePath));
        Bitmap image = retriever.getFrameAtTime(time * 1000, MediaMetadataRetriever.OPTION_CLOSEST_SYNC);
        retriever.release();
        if (image == null) {
            throw new IllegalStateException("File doesn't exist or not supported");
        }
        return image;
    }

can be called as

    Bitmap image = this.getBitmapAtTime(filePath, timeStamp, headers);

```

In this headers are ignored but it should work as of now till we have final resolution.

**Comment by souvik-ghosh** [comment]
@Janak-Nirmal Thanks for the insight! The support for `content URI` was added in @NS-BOBBY-C's PR and is available in v1.3.0. Closing this as of now.
